src/main/kotlin/no/nb/mlt/wls/domain/WLSService.kt

```diff
@@ -248,11 +229,28 @@ class WLSService(
         return createdOrder
     }
 
+    override suspend fun pickOrderItems(
+        hostName: HostName,
+        pickedItemIds: List<String>,
+        orderId: String
+    ) {
+        val order = getOrderOrThrow(hostName, orderId)
+
+        val catalogEvent =
+            transactionPort.executeInTransaction {
+                val pickedOrder = orderRepository.updateOrder(order.pickItems(pickedItemIds))
+
+                catalogEventRepository.save(OrderEvent(pickedOrder))
+            } ?: throw RuntimeException("Could not pick order items")
+
+        processCatalogEventAsync(catalogEvent)
+    }
+
     override suspend fun deleteOrder(
         hostName: HostName,
         hostOrderId: String
     ) {
-        val order = getOrderOrThrow(hostName, hostOrderId).deleteOrder()
+        val order = getOrderOrThrow(hostName, hostOrderId)
```

Comment:
```suggestion
        val order = getOrderOrThrow(hostName, hostOrderId)
        val deletedOrder = order.deleteOrder()
```
The call to remove the order should not be removed. Could be made more explicit by putting it on its own line

Reply:
Why? It does nothing

Reply:
fixed